### ISSUE TITLE
Vim mode for terminal

### ISSUE THREAD

**Issue by zbraniecki** [gnunn1/tilix#1854]
When using neovim with default navigation keys (hjkl), I can configure cmd+(hjkl) to navigate between terminals, but if one of them is a regular `zsh`, that creates a UX problem - I hold my fingers on (hjkl) when in vim and between vim terminals, but when I hit `zsh` terminal I have to reach to arrow up to get the last command and execute it again.

It would be awesome to have a mode in which when I'm in terminal, for example, `k` works like `arrow up`, and `j` works like arrow down` and `i` enters `text mode` to type my command in it.

**Comment by Laessig** [comment]
Hi. 

Your issue seems rather shell related to me. If you want to use home row navigation or other vim features in `zsh` or `bash` just activate `vi(m)` mode there.

**Comment by serafeimgr** [comment]
hello, it seems that i cannot use vim mode in tilix.
when i tried switching on gnome-terminal it works as expected.

Using the same shell (zsh)

**Comment by jordanchaitin** [comment]
Not necessarily. While it is ofc possible to enable vi mode in the shell by adding a line in the config file, it may not always be possible to do that on a remote server. Even when it is, while logging into a server using a shared account it may not always be desirable to do that. 

Furthermore, some shells like csh (default for the root user on FreeBSD) doesn't have the possibility to have  a vi mode. It is not always advisable to change root's default shell. 

So the most pragmatic way is to have a binding in the user's local terminal emulator. e.g. iTerm2 already has this as a feature.
